src/main/java/ee/carlrobert/codegpt/completions/llama/LlamaServerAgent.java

```diff
@@ -32,65 +35,94 @@ public final class LlamaServerAgent implements Disposable {
 
   public void startAgent(
       LlamaServerStartupParams params,
       ServerProgressPanel serverProgressPanel,
       Runnable onSuccess,
-      Runnable onServerTerminated) {
+      Consumer<ServerProgressPanel> onServerTerminated) {
+    this.activeServerProgressPanel = serverProgressPanel;
     ApplicationManager.getApplication().invokeLater(() -> {
       try {
-        serverProgressPanel.updateText(
+        stoppedByUser = false;
+        serverProgressPanel.displayText(
             CodeGPTBundle.get("llamaServerAgent.buildingProject.description"));
-        makeProcessHandler = new OSProcessHandler(getMakeCommandLinde());
+        makeProcessHandler = new OSProcessHandler(
+            getMakeCommandLine(params.additionalBuildParameters()));
         makeProcessHandler.addProcessListener(
-            getMakeProcessListener(params, serverProgressPanel, onSuccess, onServerTerminated));
+            getMakeProcessListener(params, onSuccess, onServerTerminated));
         makeProcessHandler.startNotify();
       } catch (ExecutionException e) {
-        throw new RuntimeException(e);
+        showServerError(e.getMessage(), onServerTerminated);
       }
     });
   }
 
   public void stopAgent() {
+    stoppedByUser = true;
+    if (makeProcessHandler != null) {
+      makeProcessHandler.destroyProcess();
+    }
     if (startServerProcessHandler != null) {
       startServerProcessHandler.destroyProcess();
     }
   }
 
   public boolean isServerRunning() {
-    return startServerProcessHandler != null
+    return (makeProcessHandler != null
+        && makeProcessHandler.isStartNotified()
+        && !makeProcessHandler.isProcessTerminated())
+        || (startServerProcessHandler != null
         && startServerProcessHandler.isStartNotified()
-        && !startServerProcessHandler.isProcessTerminated();
+        && !startServerProcessHandler.isProcessTerminated());
   }
 
   private ProcessListener getMakeProcessListener(
       LlamaServerStartupParams params,
-      ServerProgressPanel serverProgressPanel,
       Runnable onSuccess,
-      Runnable onServerTerminated) {
+      Consumer<ServerProgressPanel> onServerTerminated) {
     LOG.info("Building llama project");
 
     return new ProcessAdapter() {
+
+      private final List<String> errorLines = new CopyOnWriteArrayList<>();
+
       @Override
       public void onTextAvailable(@NotNull ProcessEvent event, @NotNull Key outputType) {
+        if (ProcessOutputType.isStderr(outputType)) {
+          errorLines.add(event.getText());
+          return;
+        }
         LOG.info(event.getText());
       }
 
       @Override
       public void processTerminated(@NotNull ProcessEvent event) {
+        int exitCode = event.getExitCode();
+        LOG.info(format("Server build exited with code %d", exitCode));
+        if (stoppedByUser) {
+          onServerTerminated.accept(activeServerProgressPanel);
+          return;
+        }
+        if (exitCode != 0) {
```

Comment:
The `make` process also prints warnings to the `stderr`, therefore the process' exit code is used to determine if there was an error